server/services/v1/observability.go

```diff
@@ -0,0 +1,203 @@
+// Copyright 2022 Tigris Data, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+
+	"github.com/fullstorydev/grpchan/inprocgrpc"
+	"github.com/go-chi/chi/v5"
+	"github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
+	"github.com/rs/zerolog/log"
+	api "github.com/tigrisdata/tigris/api/server/v1"
+	"github.com/tigrisdata/tigris/server/config"
+	"github.com/tigrisdata/tigris/server/request"
+	"google.golang.org/grpc"
+)
+
+const (
+	observabilityPattern     = "/observability/*"
+	AcceptHeader             = "Accept"
+	ApplicationJsonHeaderVal = "application/json"
+	DDApiKey                 = "DD-API-KEY"
+	DDAppKey                 = "DD-APPLICATION-KEY"
+	Query                    = "query"
+)
+
+type observabilityService struct {
+	api.UnimplementedObservabilityServer
+	ObservableProvider
+}
+
+type ObservableProvider interface {
+	QueryTimeSeriesMetrics(ctx context.Context, request *api.QueryTimeSeriesMetricsRequest) (*api.QueryTimeSeriesMetricsResponse, error)
+}
+
+type Datadog struct {
+	config.ObservabilityConfig
+}
+
+type MetricsQueryResp struct {
+	FromDate int64 `json:"from_date"`
+	ToDate   int64 `json:"to_date"`
+	Query    string
+	Series   [1]struct {
+		PointList [][2]float64 `json:"pointlist"`
+	}
+}
+
+func (dd Datadog) QueryTimeSeriesMetrics(ctx context.Context, req *api.QueryTimeSeriesMetricsRequest) (*api.QueryTimeSeriesMetricsResponse, error) {
```

Comment:
How easy it is to use this in the Quota?  The use-case is to fetch a few metrics in the quota to make a decision about the throttling. If we can just utilize this code that would be ideal. @JigarJoshi @efirs  (that may not be timeseries though)